diff --git a/tsc/internal/lsp/server.go b/tsc/internal/lsp/server.go
index f22ed6c8bd..82786c1789 100644
--- a/tsc/internal/lsp/server.go
+++ b/tsc/internal/lsp/server.go
@@ -352,6 +352,7 @@ const (
 func supportedCodeActionKinds() []lsproto.CodeActionKind {
 	return []lsproto.CodeActionKind{
 		lsproto.CodeActionKindQuickFix,
+		lsproto.CodeActionKindRefactorRewrite,
 		lsproto.CodeActionKindSourceOrganizeImportsTs,
 		lsproto.CodeActionKindSourceRemoveUnusedImportsTs,
 		lsproto.CodeActionKindSourceSortImportsTs,
